src/neo/SmartContract/ApplicationEngine.Contract.cs

```diff
@@ -28,22 +25,38 @@ partial class ApplicationEngine
         /// </summary>
         public static readonly InteropDescriptor System_Contract_CreateStandardAccount = Register("System.Contract.CreateStandardAccount", nameof(CreateStandardAccount), 0_00010000, CallFlags.None, true);
 
-        protected internal void CreateContract(byte[] script, byte[] manifest)
+        private UInt160 ContractHash(NefFile nef)
         {
-            if (script.Length == 0 || script.Length > MaxContractLength)
-                throw new ArgumentException($"Invalid Script Length: {script.Length}");
+            if (!(ScriptContainer is Transaction tx))
+                throw new InvalidOperationException();
+
+            using var script = new ScriptBuilder();
+            script.Emit(OpCode.ABORT);
+            script.EmitPush(tx.Sender);
+            script.EmitPush(nef.Script);
+
+            return script.ToArray().ToScriptHash();
+        }
```

Comment:
We also need to change the `ScriptHash` in neo-cli,
https://github.com/neo-project/neo-node/pull/676/files#diff-f3bfdce077e0db419dd0c9a5fdeb7bada0f71efa281a80230b4165af43a12b99R285 needs update.🤔

Reply:
@ProDog  Does this pr work well?

Reply:
It's works fine for me.